code-challenges.js

```diff
@@ -2,35 +2,64 @@
 
 // --------------------1) Create a function that takes in a string and returns a coded message. The coded message converts 'a' to 4, 'e' to 3, 'i' to 1, and 'o' to 0.
 
-
-
-
-
 var secretCodeWord1 = "lackadaisical"
 // Expected output: "l4ck4d41s1c4l"
 var secretCodeWord2 = "gobbledygook"
 // Expected output: "g0bbl3dyg00k"
 
-// --------------------2) Create a function that takes in an array and returns all the words that have the letter a in them.
-
-
-
+const myFunction = (string) => {
+    let newArray = string.split('');
+    let newCodedArray = newArray.map(value=> {
+        if (value == "a") {
+            return "4";
+        } else if (value == "e"){
+            return "3";
+        } else if (value == "i"){
+            return "1";
+        } else if (value == "o") {
+            return "0";
+        } else {
+            return value
+        }       
+    })
+    return newCodedArray.join('')
+}
+console.log(myFunction(secretCodeWord1))
+console.log(myFunction(secretCodeWord2))
 
+// --------------------2) Create a function that takes in an array and returns all the words that have the letter a in them.
 
 var arrayOfWords = ["Apple", "Banana", "Plum", "Cherry", "Kiwi", "Peach"]
 // Expected output: "Apple" "Banana" "Peach"
 
-
+const aFinder = (array) => {
+    let finalArray = array.filter(value=>{
+        return value.match (/[Aa]/);
+}); 
+    return finalArray;
+}    
+console.log(aFinder(arrayOfWords))
 
 
 
 // --------------------3) Create a function that takes in an array of 5 numbers and determines whether or not the array is a “full house”. A full house is one pair and one three of a kind.
 
-
-
```

Comment:
Great attempt on the full house challenge. It is a tough one. Your code works for the test variables, but it would break under a test case like [1, 1, 1, 1, 3].